earth_enterprise/src/common/tests/RunAllTests.pl

```diff
@@ -17,12 +17,24 @@
 
 use FindBin qw($Bin $Script);
 use File::Basename;
+use Cwd;
+use Cwd 'abs_path';
+
+# This script should be run from NATIVE-*-x86_64/bin/tests
+my $workingDir = abs_path(cwd());
+my $scriptDir = dirname(abs_path($0));
+my $notInBuildDir = (index($scriptDir, "NATIVE") == -1);
```

Comment:
Minor: Since Perl is famous for regular expressions, or regular expressions are famous because of Perl, you could use something like:

```Perl
my $notInBuildDir = $scriptDir !~ /NATIVE-[a-zA-Z0-9_-]*\/bin\/tests$/;
```

This should work in more cases than `(index($scriptDir, "NATIVE") == -1)`.